packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButton.tsx

```diff
@@ -266,13 +275,15 @@ export const useSpinButton_unstable = (props: SpinButtonProps, ref: React.Ref<HT
 
     if (valueChanged || displayValueChanged) {
       onChange?.(e, { value: roundedValue, displayValue: newDisplayValue });
+    } else if (!valueChanged && newValue === null) {
+      onChange?.(e, { value: null });
     }
```

Comment:
Will this fire on all blurs, ENTER keydown, any attempts to step (if so, is that what we want)?

Reply:
It does and it was intentional. My thinking was that `null` is a special case so SpinButton should fire events when a user tries to change it but thinking about it more, I'm not so sure. `null` has to be set from the app, SpinButton never sets the value to `null` internally, so the app will be fully aware of this state. I'm thinking I'll remove this special case.

What do you think?

Reply:
I don't have a strong opinion, but my gut instinct is that it'd be nice if the firing of `onChange` was consistent

Reply:
Agreed. Based on our conversation [here](https://github.com/microsoft/fluentui/pull/23004#discussion_r874289988) I'm going to update this so it only fires when the value actually changes.